Source/Orts.Simulation/Simulation/Signalling/Signals.cs

```diff
@@ -8308,7 +8338,7 @@ public enum HoldState                // signal is locked in hold
 
         public Train.TCSubpathRoute signalRoute = new Train.TCSubpathRoute();  // train route from signal
         public int trainRouteDirectionIndex;    // direction index in train route array (usually 0, value 1 valid for Manual only)
-        private int thisTrainRouteIndex;        // index of section after signal in train route list
+        public int thisTrainRouteIndex;        // index of section after signal in train route list
```

Comment:
not a good practice to have public fields (and this is wide spread throughout the whole solution).
If we make this a public field, should ast least follow naming conventions (camel casing, start with upper case letter). 
More generally I'd suggest some refactoring separately to cleanup code.